packages/browser-wallet/src/popup/pages/Identity/Identity.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect } from 'react';
+import { useAtom } from 'jotai';
+import { selectedIdentityAtom } from '@popup/store/identity';
+import IdCard from '@popup/shared/IdCard';
+import { IdentityStatus } from '@shared/storage/types';
+import { AttributeKey } from '@concordium/web-sdk';
+import attributeNames from 'wallet-common-helpers/constants/attributeNames.json';
+import { formatAttributeValue, compareAttributes } from 'wallet-common-helpers';
+import { getIdObject } from '@shared/utils/identity-helpers';
+
+export default function Identity() {
+    const [selectedIdentity, updateSelectedIdentity] = useAtom(selectedIdentityAtom);
+
+    useEffect(() => {
+        // TODO: Check identity status in background
+        if (selectedIdentity?.status === IdentityStatus.Pending) {
+            getIdObject(selectedIdentity.location).then((response) => {
+                if (response.error) {
+                    updateSelectedIdentity({
```

Comment:
You should not update the selected identity if the selected has changed in the meantime.

Reply:
Yeah, You are correct. I do rework the confirmation in a later PR, so I'd prefer leaving it like this/removing this instead.